### ISSUE TITLE
Have Microclimate deal with residues interception of precipitation and radiation

### ISSUE THREAD

**Issue by HamishBrownPFR** [APSIMInitiative/ApsimX#1196]
I have just written some crappy code to get evaporation into the multi pore water model.  I think SoilWat and Swim have their own internal calculations for doing this also.  It would make sense to have a single method for calculating potential soil evaporation from the soil located in micro climate rather than duplicating this in every soil water model.  
It would also tidy things up if we had microclimate deal with the effects of residues on interception of radiation and precipitation in microclimate as this is not handled well at the moment.
I am happy to implement some stuff here but wanted to catch other peoples thoughts first.

Please discuss
@hut104 @hol353 @zur003 @rcichota @sno036

**Comment by hut104** [comment]
I think it makes sense for micromet to calculate the potential (Eos) value.  We did talk about this some time back.  SWIM and Soilwat used to use the same code (duplicated) to do this (a bit silly really).

This may not be a simple job though – be ready for it to take longer than you expect.

**Comment by sno036** [comment]
PS - do I have to add all the @'s in the comments?

**Comment by HamishBrownPFR** [comment]
I agree, but it is also good to fix these things when we are thinking about them and good to fix them now before APSIMX has much of a user base.  

What about the residual interception does any one have thoughts on that.

you don't have to do the @'s if they are there once then they get notified of the thread

**Comment by sno036** [comment]
I've just talked to Neil about this.

Dean - if Russell and I move the EOS calculation from SoilWater to MicroMet will that affect what you're doing with refactoring SoilWater?


On 20/09/2016, at 5:15 PM, Hamish Brown <notifications@github.com<mailto:notifications@github.com>> wrote:


I agree, but it is also good to fix these things when we are thinking about them and good to fix them now before APSIMX has much of a user base.

What about the residual interception does any one have thoughts on that.

you don't have to do the @'s if they are there once then they get notified of the thread

�

**Comment by HamishBrownPFR** [comment]
@sno036   Some progress has been made toward resolving this issue but some discussion is needed to determine some of the details.
- how do we want to model the proportion of precip intercepted.  
- Should different crops and residues have different coefficients for the amount of precipitation they intercept.
- How do we want to deal with height of lying residues  Currently lying residue pools have a height of 50mm and we have problems is the pools become shorter than this.  
I will try to organize a skype meeting with us and @hut104 so we can update you on where we got to and get you involved in decision making for how we want to implement this.
